addons/product/models/res_company.py

```diff
@@ -7,24 +7,28 @@
 class ResCompany(models.Model):
     _inherit = "res.company"
 
-    @api.model
-    def create(self, vals):
-        new_company = super(ResCompany, self).create(vals)
+    @api.model_create_multi
+    def create(self, vals_list):
+        companies = super(ResCompany, self).create(vals_list)
         ProductPricelist = self.env['product.pricelist']
-        pricelist = ProductPricelist.search([('currency_id', '=', new_company.currency_id.id), ('company_id', '=', False)], limit=1)
-        if not pricelist:
-            params = {'currency': new_company.currency_id.name}
-            pricelist = ProductPricelist.create({
-                'name': _("Default %(currency)s pricelist") %  params,
-                'currency_id': new_company.currency_id.id,
-            })
-        self.env['ir.property']._set_default(
-            'property_product_pricelist',
-            'res.partner',
-            pricelist,
-            new_company,
-        )
-        return new_company
+        for new_company in companies:
+            pricelist = ProductPricelist.search([
+                ('currency_id', '=', new_company.currency_id.id),
+                ('company_id', '=', False)
+            ], limit=1)
+            if not pricelist:
+                params = {'currency': new_company.currency_id.name}
+                pricelist = ProductPricelist.create({
+                    'name': _("Default %(currency)s pricelist") %  params,
+                    'currency_id': new_company.currency_id.id,
+                })
```

Comment:
<pre>
currency_companies = self.env['res.company'].read_group([
    ('id', 'in', companies.ids),
], ['currency_id', 'company_ids:array_agg(id)'], ['currency_id'])
for vals in currency_companies:
    currency_id = vals['currency_id'][0]
    pricelist = ProductPricelist.search([
        ('currency_id', '=', currency_id),
        ('company_id', '=', False),
    ], limit=1)
    if not pricelist:
        currency_name = self.env['res.currency'].browse(currency_id)
        for new_company in self.browse(vals['company_ids']):
            params = {'currency': currency_name}
            pricelist = ProductPricelist.create({
                'name': _("Default %(currency)s pricelist") %  params,
                'currency_id': currency_id,
            })
</pre>